fun_3000/wrangling/boost_corpus_with_ontology.py

```diff
@@ -0,0 +1,44 @@
+import io
+import optparse
+from os import path
+import shutil
+
+from utils import PARENT_DIR, read_source
+
+
+def run(run_directory, corpus_filename="output.txt", multiplier=1):
+    """
+    Takes a corpus file and appends ontology user-specified number of times. Output filename is of format [MULTIPLIER]_boost_[CORPUS_FILENAME] (e.g. filename 'output.txt' and multiplier `5` would result in output filename 5_boost_output.txt)
+
+    :param run_directory: where the data for the given run is stored (e.g., run_1, run_2)
+    :param corpus_filename: what the clean corpus file is called within the run directory
+    :param multiplier: int by which to multiply the available ontologies
+    :return:
+    """
+    # figure out where the corpus file is. should be in ddl_nlp/data/{run_directory}/{corpus_filename}
+    absolute_corpus_filename = path.join(PARENT_DIR, 'data', run_directory, corpus_filename)
+
+    # construct filename based on input corpus name and multiplier
+    output_filename = path.join(PARENT_DIR, 'data', run_directory, "%s_boost_%s" % (multiplier, corpus_filename))
+
+    # copy corpus file data into output file
+    shutil.copy(absolute_corpus_filename, output_filename)
+
+    # get our ontology file
+    ontology = read_source(run_directory, source_type='ontology')
```

Comment:
jk the real thing is that we want to instead of compile all ontologies at once, open each ontology in a for loop and append its multiplied version separately. We think this will be more performant. This still involves multiplying the string read in from a single ontology into a bigger string so not sure that's gonna be great.

There is also a world where we might figure out there is a better streaming (perhaps with `bash`) option that we can use here but we are going to go with the stated python implementation above for now.